protocols/ntree/simulation.go

```diff
@@ -0,0 +1,68 @@
+package ntree
+
+import (
+	"github.com/BurntSushi/toml"
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/dedis/cothority/lib/monitor"
+	"github.com/dedis/cothority/lib/sda"
+)
+
+func init() {
+	sda.SimulationRegister("NaiveTreeSimulation", NewSimulation)
+	sda.ProtocolRegisterName("CosiNtree", func(node *sda.Node) (sda.ProtocolInstance, error) { return NewProtocol(node) })
+}
+
+type Simulation struct {
+	sda.SimulationBFTree
+}
+
+func NewSimulation(config string) (sda.Simulation, error) {
+	es := &Simulation{}
+	_, err := toml.Decode(config, es)
+	if err != nil {
+		return nil, err
+	}
+	return es, nil
+}
+
+func (e *Simulation) Setup(dir string, hosts []string) (
+	*sda.SimulationConfig, error) {
+	sc := &sda.SimulationConfig{}
+	e.CreateEntityList(sc, hosts, 2000)
+	err := e.CreateTree(sc)
+	if err != nil {
+		return nil, err
+	}
+	return sc, nil
+}
+
+func (e *Simulation) Run(config *sda.SimulationConfig) error {
+	msg := []byte("Helle Ntree")
+	size := config.Tree.Size()
+	dbg.Lvl2("Size is:", size, "rounds:", e.Rounds)
+	for round := 0; round < e.Rounds; round++ {
+		dbg.Lvl1("Starting round", round)
+		round := monitor.NewMeasure("round")
+
+		node, err := config.Overlay.NewNodeEmptyName("CosiNtree", config.Tree)
+		if err != nil {
+			dbg.Error("Quitting the simuation....", err)
+			return err
+		}
+		proto, err := NewRootProtocol(msg, node)
+		if err != nil {
+			dbg.Error("Quitting the simuation....", err)
+			return err
+		}
+		node.SetProtocolInstance(proto)
+		err = proto.Start()
+		if err != nil {
```

Comment:
No, it doesn't start. `StartNewNodeName` calls `CreateNewNodeName` and the calls `Start` on the created node. So if you only call `CreateNewNodeName` you have the node, the ProtocolInstance, but no `Start` yet. This is why I don't like the `NewNodeEmptyName` and would like to get rid of it.